build/scripts/do-release.sh

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env bash
+
+set -e
+
+cd "$(dirname "$0")/../.." || exit
+
+usage() {
+  echo "Usage: $(basename "$0") [-h|--help]
+where :
+  -h| --help Display this help text
+" 1>&2
+  exit 1
+}
+
+if [[ ($# -ne 0) || ( $1 == "--help") ||  $1 == "-h" ]]; then
+  usage
+fi
+
+readReleaseProp() {
+  grep "^$1=" release.properties | head -n1 | sed "s/$1=//"
+}
+
+readPomVersion() {
+  grep '^    <version>' "$1" | head -n1 | sed -E 's|.*<version>(.*)</version>.*|\1|'
+}
+
+copyReleaseArtifacts() {
+  while IFS= read -r -d '' file; do
+    pushd "$(dirname "$file")" >/dev/null
+    gpg --armor --detach-sign "$(basename "$file")"
+    sha256sum "$(basename "$file")" > "$(basename "$file").sha256"
+    popd >/dev/null
+    mv "$file"{,.sha256,.asc} "$RELEASE"
+  done < <(find geomesa-* -name '*-bin.tar.gz' -print0)
+}
+
+JAVA_VERSION="$(mvn help:evaluate -Dexpression=jdk.version -q -DforceStdout)"
+if ! [[ $(java -version 2>&1 | head -n 1 | cut -d'"' -f2) =~ ^$JAVA_VERSION.* ]]; then
+  echo "Error: invalid Java version - Java $JAVA_VERSION required"
+  exit 1
+fi
+
+if ! [[ $(which gpg) ]]; then
+  echo "Error: gpg executable not found"
+  exit 1
+fi
+
+# get current branch we're releasing off
+BRANCH="$(git branch --show-current)"
+
+# use the maven release plugin to prep the pom changes but use dryRun to skip commit and tag
+mvn release:prepare \
+  -DdryRun=true \
+  -DautoVersionSubmodules=true \
+  -Darguments="-DskipTests -Dmaven.javadoc.skip=true -Ppython" \
+  -Ppython
+
+RELEASE="$(readPomVersion pom.xml.tag)"
+TAG="$(readReleaseProp scm.tag)"
+NEXT="$(readPomVersion pom.xml.next)"
+
+# update README versions and commit
+for pom in pom.xml pom.xml.tag pom.xml.next; do
+  sed -i "s|<geomesa\.release\.version>.*|<geomesa.release.version>$RELEASE</geomesa.release.version>|" "$pom"
+  sed -i "s|<geomesa\.devel\.version>.*|<geomesa.devel.version>$NEXT</geomesa.devel.version>|" "$pom"
+done
+mvn clean install -pl .
+git commit -am "Set version for release $RELEASE"
+
+# commit release tag
+find . -name pom.xml -exec mv {}.tag {} \;
+git commit -am "[maven-release-plugin] prepare release $TAG"
```

Comment:
Are we keeping `[maven-release-plugin]` just for consistency? Seems odd since the message didn't come from the release plugin.

Reply:
you ready to do the next release now? ;)

Reply:
happy to try :D